addons/event_sale/tests/test_event_sale.py

```diff
@@ -376,7 +384,200 @@ def test_ticket_price_with_pricelist_and_tax(self):
+                              product_lst_price, ticket_price, qty,
+                              rule_type, rule_amount, min_qty,
+                              rule_based_on=None, product_standard_price=10,
+                              price_list_currency=None):
+            """Create a sale order and related instances based on the provided configuration.
+
+            :param bool is_product_templ: whether to use a product template or a product
+            :param float product_lst_price: price of the product linked to the event ticket
+            :param float ticket_price: price of the ticket
+            :param int qty: number of ticket ordered in the sale order line
+            :param str rule_type: price list item compute_price (formula, fixed or percentage)
+            :param float rule_amount: amount used for the rule_type (ex.: rule_type = fixed, it means the fixed price)
+            :param int min_qty: minimum quantity for the rule to be applied
+            :param str rule_based_on: by default on the ticket price (list_price), can be set to standard_price (cost)
+            :param float product_standard_price: product standard_price (product cost which is used as ticket cost)
+            :param res_currency price_list_currency: currency of the price list
+            :return: (sale_order, sale_order_line)
+            """
+            if price_list_currency:
+                price_list.write({'currency_id': price_list_currency.id})
+            if is_product_templ:
+                product_templ = self.env['product.template'].create({
+                    'name': 'event_product',
+                    'detailed_type': 'event',
+                    'standard_price': product_standard_price,
+                    'list_price': product_lst_price,
+                    'taxes_id': [Command.set([tax.id])],
+                })
+            else:
+                product = self.env['product.product'].create({
+                    'name': 'event_product',
+                    'detailed_type': 'event',
+                    'lst_price': product_lst_price,
+                    'standard_price': product_standard_price,
+                    'taxes_id': [Command.set([tax.id])],
+                })
+            price_list.item_ids += self.env['product.pricelist.item'].create({
+                'compute_price': rule_type,
+                'base': rule_based_on if rule_based_on else 'list_price',
+                'price_discount': rule_amount if rule_type == 'formula' else 0.0,
+                'fixed_price': rule_amount if rule_type == 'fixed' else 0.0,
+                'percent_price': rule_amount if rule_type == 'percentage' else 0.0,
+                'min_quantity': min_qty,
+                'product_id': product.id if not is_product_templ else None,
+                'product_tmpl_id': product_templ.id if is_product_templ else None,
+                'applied_on': '1_product' if is_product_templ else '0_product_variant',
+            })
```

Comment:
This is a bit dense. In this case I would go for a `item_values` dict construction with only one `if` per test, like below.
(The part depending on `is_product_templ` can be handled as above).
```suggestion
            pricelist_item_values = {
                'base': rule_based_on if rule_based_on else 'list_price',
                'compute_price': rule_type,
                'fixed_price': 0.,
                'price_discount': 0.,
                'min_quantity': min_qty,
                'percent_price': 0.,
            }
            rule_field = {
                'fixed': 'fixed_price',
                'percentage': 'percent_price',
                'formula': 'discount_price',
            }[rule_type]
            pricelist_item_values[rule_field] = rule_amount
```

Reply:
Field mapping is a good idea. I have reworked the code following this idea.